### ISSUE TITLE
Problem with the NPM package - 403 Forbidden from dl.bintray.com

### ISSUE THREAD

**Issue by durkiewicz** [elm/compiler#2202]
**Quick Summary:** Cannot install Elm 0.16.0

Unfortunately, I have a dependency to Elm 0.16.0 in my package.json. When I run `npm ci` (or `npm install`), I'm getting:

```sh
> node install.js

Unable to download elm-reactor assets. elm-reactor may not work properly.
403
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! elm@0.16.0 install: `node install.js`
npm ERR! Exit status 1
```

The reason seems to be quite obvious: the requests to the URLs below end up with 403 Forbidden errors.

https://dl.bintray.com/elmlang/elm-platform/0.16.0/linux-x64.tar.gz
https://dl.bintray.com/elmlang/elm-platform/0.16.0/elm-reactor-assets.tar.gz

Can you please fix it?

I'm raising the bug here as the "Issues" link from https://www.npmjs.com/package/elm points here.

**Comment by lydell** [comment]
Hi!

See this Discourse thread for more information: https://discourse.elm-lang.org/t/psa-old-elm-binary-downloads-now-gone/7319

**Comment by soenkehahn** [comment]
The discourse thread mentions workarounds for `0.18.0`. I have the same problem with `0.17.1`, and those workarounds don't work for that version. For example, I haven't been able to find any binary tarballs on github for `0.17.1`. ([here](https://github.com/elm-lang/elm-platform/releases)) Are there any known workaround for `0.17.1`?

I was looking at the tags on https://github.com/elm-lang/elm-platform, hoping that I could build `elm-platform` from source. But it doesn't have a `0.17.1` tag. Which repo(s) was `0.17.1` originally built from? And from which commits?

**Comment by lydell** [comment]
The Discourse post links to a place to download many old Elm binaries, including 0.17.1. So you can download from there if you trust me – that’s a workaround for 0.17.1.

Elm 0.18.0 seems to be the only pre-0.19 version that was updated to use GitHub releases instead of bintray.

(I don’t know which repo(s) 0.17.1 was originally built from.)

**Comment by avh4** [comment]
I'm not sure if this will lead to a solution for 0.17, but I believe this is around the commit that would be relevant to the 0.17.1 release: https://github.com/elm-lang/elm-platform/tree/b7a68b23dc3db5c0c7707d70955e953b8cada7f1

**Comment by soenkehahn** [comment]
@avh4: Yeah, I looked at some of those untagged commits. The build instructions say that I need a `cabal` version that supports the `sandbox` sub-command. Which was removed from `cabal` some time ago, it seems: https://github.com/haskell/cabal/issues/6445. I gave up at that point.

**Comment by evancz** [comment]
Some people use `npm` to get their Elm binaries, but it is more direct to use `curl` as described [here](https://github.com/elm/compiler/blob/main/installers/linux/README.md).

Practically speaking, we will not be able to prioritize working on the `npm` installer, but the binaries themselves should still work just fine. So my recommendation would be to distribute the particular binary that you want using `curl` or something along those lines.
